### PR TITLE
Refactor Workpiece.getAll…() methods

### PR DESCRIPTION
**Pull request by matthias-ronge**
This looked odd, and refactoring it makes the code (and possible bugs) easier to spot.

### CODE REVIEW

**Comment by Kathrin-Huber** [comment]
Please resolve conflicts

**Comment by solth** [comment]
@matthias-ronge please rebase against current master

File: Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java [hunk 1]
```diff
@@ -180,42 +178,41 @@ public boolean equals(Object o) {
...
      *
-     * @return list of all media units with type "page", sorted by their "ORDER" attribute.
+     * @return all media units with type "page", sorted by their {@code order}
      */
-    public List<MediaUnit> getAllMediaUnitsSorted() {
-        List<MediaUnit> mediaUnits = getAllMediaUnits();
+    public List<MediaUnit> getAllMediaUnitsFilteredByTypePageAndSorted() {
+        List<MediaUnit> mediaUnits = treeStream(mediaUnit).filter(m -> Objects.equals(m.getType(), MediaUnit.TYPE_PAGE))
+                .collect(Collectors.toList());
         mediaUnits.sort(Comparator.comparing(MediaUnit::getOrder));
```
Comment (solth) [inline Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java:205]:
```suggestion
                .sorted(Comparator.comparing(MediaUnit::getOrder)).collect(Collectors.toList());
```

File: Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java [hunk 2]
```diff
@@ -239,4 +236,16 @@ public boolean equals(Object o) {
...
+    /**
+     * Generates a stream of nodes from structure tree.
+     *
+     * @param tree
+     *            starting node
+     * @return all nodes as stream
+     */
+    @SuppressWarnings("unchecked")
+    public static <T extends Parent<T>> Stream<T> treeStream(Parent<T> tree) {
+        return Stream.concat(Stream.of((T) tree), tree.getChildren().stream().flatMap(child -> treeStream(child)));
```
Comment (solth) [inline Kitodo-API/src/main/java/org/kitodo/api/dataformat/Workpiece.java:249]:
```suggestion
        return Stream.concat(Stream.of((T) tree), tree.getChildren().stream().flatMap(Workpiece::treeStream));
```

**Pull request merged**